File: KampIntro_Odevler/Gun_05_Odev_05/Gun_05_Odev_05/Concrete/SaleManager.cs
using Gun_05_Odev_05.Abstract;
using Gun_05_Odev_05.Entities;
using System;
using System.Collections.Generic;
using System.Text;

namespace Gun_05_Odev_05.Concrete
{
    public class SaleManager : BaseSaleManager
    {
        IPlayerExistService _playerExistService;
        ICampaignExistService _campaignExistService;


        public SaleManager(IPlayerExistService playerExistService)
        {
            _playerExistService = playerExistService;
        }
        public override void Add(Sale sale, Player player)
        {

            if (_playerExistService.CheckIfPlayerExist(player) == true)
            {
                    base.Add(sale, player);
                    Console.WriteLine("Kampanyalı satış gerçekleştirildi");
                    Console.ReadLine();
                }
            else
            {
                Console.WriteLine("Satış yapılması istenen oyuncu kayıtlı değil");
                Console.ReadLine();
            }
        }
    }
}

File: Java_React_Ödevler/Java_React_Ödev_Gün_04_Ödev_03_OyunSatis/Gun_04_Odev_03/Concrete/SaleManager.java

public class SaleManager extends BaseSaleManager {

	PlayerExistService playerExistService;
    CampaignExistService campaignExistService;


    public SaleManager(PlayerExistService playerExistService)
    {
        this.playerExistService = playerExistService;
    }
    public void add(Sale sale, Player player)
    {

        if (this.playerExistService.CheckIfPlayerExist(player) == true)
        {
                super.add(sale, player);
                System.out.println("Kampanyalý satýþ gerçekleþtirildi" + " " + player.getFirstName() + " " +
                															   player.getLastName()  + " " +
                															   sale.getName()		 + " " +
                															   sale.getDiscountRate());
                
            }
        else
        {
        	System.out.println("Satýþ yapýlmasý istenen oyuncu kayýtlý deðil");
            
        }
    }
}